### PR TITLE
Add serverless mode for independent Task execution

### PR DESCRIPTION
**Pull request by aarontp**
This will allow us to execute one-off Tasks without any Turbinia or Cloud infrastructure.  This can be used to make the Tasks more reusable outside of the framework or just for testing.  This is what it looks like:

```
$ turbiniactl -f --run_local --task PlasoTask rawdisk -l ~/tmp/turbinia/images/test-small.img
[INFO] Creating request c954257c28c7481ab1a784a4e5efdaae with evidence /home/aaronpeterson/tmp/turbinia/images/test-small.img
[DEBUG] --run_local specified so not sending request to server
[DEBUG] Looking up Task PlasoTask by name
[INFO] Running Task PlasoTask locally
[INFO] Starting Task PlasoTask 3e1c1a7d08c040269a56d6fcddd077de
[INFO] Creating new directory /var/tmp/1540595266-3e1c1a7d08c040269a56d6fcddd077de-PlasoTask
[INFO] Running plaso as [log2timeline.py --status_view none --hashers all --partition all --vss_stores all --logfile /var/tmp/1540595266-3e1c1a7d08c040269a56d6fcddd077de-PlasoTask/3e1c1a7d08c040269a56d6fcddd077de.log /var/tmp/1540595266-3e1c1a7d08c040269a56d6fcddd077de-Pl
asoTask/3e1c1a7d08c040269a56d6fcddd077de.plaso /usr/local/google/home/aaronpeterson/tmp/turbinia/images/test-small.img]
Checking availability and versions of dependencies.
[OPTIONAL]      missing: lzma.
[OK]

Source path     : /home/aaronpeterson/tmp/turbinia/images/test-small.img
Source type     : storage media image
Processing time : 00:00:00

Processing started.
Processing completed.

[INFO] Output file at /var/tmp/1540595266-3e1c1a7d08c040269a56d6fcddd077de-PlasoTask/3e1c1a7d08c040269a56d6fcddd077de.log
[INFO] Completed successfully in 0:00:11.005304 on aaronpeterson
[DEBUG] Result close successful. Status is [Completed successfully in 0:00:11.005304 on aaronpeterson]
[DEBUG] Checking TurbiniaTaskResult for serializability
[INFO] Result check: Successful
[DEBUG] Returning original result object 0c7cf1ac454e413eaf7c25f3087e4506 after task execution
[INFO] Task execution result: {'_log': [u'Running plaso as [log2timeline.py --status_view none --hashers all --partition all --vss_stores all --logfile /var/tmp/1540595266-3e1c1a7d08c040269a56d6fcddd077de-PlasoTask/3e1c1a7d08c040269a56d6fcddd077de.log /var/tmp/1540595266-
3e1c1a7d08c040269a56d6fcddd077de-PlasoTask/3e1c1a7d08c040269a56d6fcddd077de.plaso /home/aaronpeterson/tmp/turbinia/images/test-small.img]',
          u'Output file at /var/tmp/1540595266-3e1c1a7d08c040269a56d6fcddd077de-PlasoTask/3e1c1a7d08c040269a56d6fcddd077de.log',
          u'Completed successfully in 0:00:11.005304 on aaronpeterson'],
 'base_output_dir': u'/var/tmp',
 'closed': True,
 'error': {u'stderr': None, u'stdout': None},
 'evidence': [PlasoFile:PlasoFile:/var/tmp/1540595266-3e1c1a7d08c040269a56d6fcddd077de-PlasoTask/3e1c1a7d08c040269a56d6fcddd077de.plaso],
 'id': '0c7cf1ac454e413eaf7c25f3087e4506',
 'input_evidence': [RawDisk:/home/aaronpeterson/tmp/turbinia/images/test-small.img:/home/aaronpeterson/tmp/turbinia/images/test-small.img],
 'output_dir': u'/var/tmp/1540595266-3e1c1a7d08c040269a56d6fcddd077de-PlasoTask',
 'request_id': 'c954257c28c7481ab1a784a4e5efdaae',
 'run_time': datetime.timedelta(0, 11, 5304),
 'saved_paths': [u'/var/tmp/1540595266-3e1c1a7d08c040269a56d6fcddd077de-PlasoTask/3e1c1a7d08c040269a56d6fcddd077de.plaso'],
 'start_time': datetime.datetime(2018, 10, 26, 16, 7, 47, 344438),
 'status': u'Completed successfully in 0:00:11.005304 on aaronpeterson.svl.corp.google.com',
 'successful': True,
 'task_id': '3e1c1a7d08c040269a56d6fcddd077de',
 'task_name': 'PlasoTask',
 'user': 'aaronpeterson',
 'worker_name': 'aaronpeterson'}
[INFO] Done.
```

### CODE REVIEW

File: turbinia/workers/__init__.py [hunk 1]
```diff
@@ -119,7 +125,8 @@ def close(self, task, success, status=None):
     for evidence in self.evidence:
       if evidence.local_path:
         self.saved_paths.append(evidence.local_path)
-        if evidence.copyable and not config.SHARED_FILESYSTEM:
+        if (evidence.copyable and not config.SHARED_FILESYSTEM
```
Comment (berggren) [inline turbinia/workers/__init__.py]: Little hard to follow this. Maybe:
```
if not task_run_local:
  if evidence.copyable and not config.SHARED_FILESYSTEM:
```

File: turbinia/workers/__init__.py [hunk 2]
```diff
@@ -325,7 +336,8 @@ def setup(self, evidence):
           request_id=self.request_id)
     self.output_dir = self.result.output_dir
 
-    if evidence.copyable and not config.SHARED_FILESYSTEM:
+    if (not self.run_local and evidence.copyable and not
```
Comment (berggren) [inline turbinia/workers/__init__.py]: Same here:
```
if not self.run_local:
  if evidence.copyable and not config.SHARED_FILESYSTEM:
```

File: turbinia/turbiniactl.py [hunk 3]
```diff
@@ -311,11 +322,24 @@ def main():
...
     else:
       client = TurbiniaClient()
   else:
     client = None
 
+  server_flags_set = args.server or args.use_celery or args.command == 'server'
+  worker_flags_set = (
+      args.use_celery or args.command in ('psqworker', 'celeryworker'))
+  if args.run_local and (server_flags_set or worker_flags_set):
+    log.warning('--run_local flag is not compatible with server/worker flags')
```
[the remark this answers is not in the document]
Comment (aarontp) [reply turbinia/turbiniactl.py]: Yeah, thanks.  Done.

File: turbinia/turbiniactl.py [hunk 4]
```diff
@@ -311,11 +322,24 @@ def main():
...
+    sys.exit(1)
+
+  if args.run_local and not args.task:
+    log.warning('--run_local flag requires --task flag')
```
[the remark this answers is not in the document]
Comment (aarontp) [reply turbinia/turbiniactl.py]: Done.

File: turbinia/workers/__init__.py (hunk 1 above)
Comment (aarontp) [reply turbinia/workers/__init__.py]: Sure, Done.

File: turbinia/workers/__init__.py (hunk 2 above)
Comment (aarontp) [reply turbinia/workers/__init__.py]: Done.

File: turbinia/turbiniactl.py [hunk 5]
```diff
@@ -477,6 +505,16 @@ def main():
           request_id=request.request_id,
           all_fields=args.all_fields))
 
+  if args.run_local and not evidence_:
+    log.warning('Evidence must be specified if using --run_local')
+    sys.exit(1)
+  if args.run_local and evidence_.cloud_only:
+    log.warning('--run_local cannot be used with Cloud only Evidence types')
```
[the remark this answers is not in the document]
Comment (aarontp) [reply turbinia/turbiniactl.py]: Done.

File: turbinia/turbiniactl.py [hunk 6]
```diff
@@ -477,6 +505,16 @@ def main():
           request_id=request.request_id,
           all_fields=args.all_fields))
 
+  if args.run_local and not evidence_:
+    log.warning('Evidence must be specified if using --run_local')
```
[the remark this answers is not in the document]
Comment (aarontp) [reply turbinia/turbiniactl.py]: Done.
